scripts/tests/satellites-integration.sh

```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+set -euo pipefail # don't use -x as it will leak the mirror credentials
+
+# to run this locally; in the root of the repo:
+#   ./earthly +earthly-docker && EARTHLY_IMAGE="earthly/earthly:dev-$(git rev-parse --abbrev-ref HEAD | sed 's/\//_/g')" scripts/tests/satellites-integration.sh
+
+FRONTEND=${FRONTEND:-docker}
+EARTHLY_IMAGE=${EARTHLY_IMAGE:-earthly/earthly:dev-main}
```

Comment:
instead of using the earthly image, I think it's better to just use the earthly cli directly. This is why I made a follow up comment: 

> It might be better to make a completely new satellite-ingration.sh test; you can reuse the script setup parts from https://github.com/earthly/earthly/blob/main/scripts/tests/cloud-docker-credentials-integration.sh

Using the earthly image test makes it difficult since state (config files) are not stored between runs. e.g.

`"$FRONTEND" run --rm --privileged -e EARTHLY_TOKEN="${EARTHLY_TOKEN}" --entrypoint="" "${EARTHLY_IMAGE}" /bin/sh -c "earthly org select earthly-technologies"` does not persist the select org to following commands, which doesn't provide the test coverage that's needed.

Reply:
here's a more complete example:

```
#!/usr/bin/env bash
set -eu

earthly=${earthly:=earthly}
earthly=$(realpath "$earthly")
echo "running tests with $earthly"

# prevent the self-update of earthly from running (this ensures no bogus data is printed to stdout,
# which would mess with the secrets data being fetched)
date +%s > /tmp/last-earthly-prerelease-check

# ensure earthly login works (and print out who gets logged in)
test -n "$EARTHLY_TOKEN"
"$earthly" account login

# test --org
"$earthly" sat --org <org> --sat <sat> inspect

# test EARTHLY_ORG env
EARTHLY_ORG=ORG "$earthly" sat --sat <sat> inspect

# test org select
"$earthly" org select <foo>
"$earthly" sat --sat <sat> inspect
```

Reply:
Good point. It only seemed to be working beforehand because it was relying on the "org guessing" logic. Working on it.

Reply:
It might be a good idea to `2>&1 | tee output` then make sure `Auto-selecting the default org` doesn't show up in the output.

Reply:
I tried adding the check for `Auto-selecting the default org` but it was getting a bit too verbose for my liking and we want to remove that auto-selection soon anyways. But the new tests pass (thank you for helping write those!) and they are hooked-in to CI.